### PR TITLE
Merge master into dev.major

### PR DESCRIPTION
**Pull request by Ericgig**
**Description**
Merge `master` as of 4.7.1, into `dev.major`.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -73,11 +74,33 @@ jobs:
...
+          # Ignore DeprecationWarnings raised by versions of
+          # setuptools >= 65.0.0 during pyximport imports This can be removed
+          # once https://github.com/cython/cython/issues/4985
+          # is fixed and released.
           - case-name: Python 3.10
             os: ubuntu-latest
             python-version: "3.10"
             condaforge: 1
-            pytest-extra-options: "-W ignore::ImportWarning -W ignore::DeprecationWarning:cvxpy.interface.scipy_wrapper"
+            pytest-extra-options: "-W ignore::ImportWarning -W ignore::DeprecationWarning:cvxpy.interface.scipy_wrapper -W ignore:Absolute:DeprecationWarning"
```
Comment (hodgestar) [inline .github/workflows/tests.yml:85]: We should see what warning ignores we can drop here, but definitely for later PRs.

File: doc/guide/heom/intro.rst [hunk 2]
```diff
@@ -40,4 +40,4 @@ In addition to support for bosonic environments, QuTiP also provides support for
 feriomic environments which is described in :doc:`fermionic`.
 
 Both bosonic and fermionic environments are supported via a single solver,
-:class:`HEOMSolver`, that supports solving for both dynamics and steady-states.
+:class:`~qutip.nonmarkov.heom.HEOMSolver`, that supports solving for both dynamics and steady-states.
```
Comment (hodgestar) [inline doc/guide/heom/intro.rst:43]: We need to fix this link, but happy for that to happen in a follow up PR unless we have to do it in this one for some reason.

File: .github/workflows/build_documentation.yml [hunk 3]
```diff
@@ -31,16 +33,32 @@ jobs:
...
+        uses: actions/upload-artifact@v3
+        with:
+          name: qutip_pdf_docs
+          path: doc/_build/latex/*
+          if-no-files-found: error
+
+      - name: Build HTML documentation
+        working-directory: doc
+        run: |
+          make html SPHINXOPTS="--keep-going -T"
```
[the remark this answers is not in the document]
Comment (Ericgig) [reply .github/workflows/build_documentation.yml]: I will try it. I had some problem with failing latex build and we have a lot of missing link warnings, but it looks like they are tolerated.

**Comment by Ericgig** [comment]
I also deactivated towncrier in this that we need to remember to add back for the full release.

**Pull request merged**